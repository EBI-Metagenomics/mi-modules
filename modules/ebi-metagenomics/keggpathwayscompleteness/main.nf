
process KEGGPATHWAYSCOMPLETENESS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/kegg-pathways-completeness:1.0.5--pyhdfd78af_0':
        'biocontainers/kegg-pathways-completeness:1.0.5--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(filtered_tbl), path(ko_list)

    output:
    tuple val(meta), path("*_contigs.tsv") , emit: kegg_contigs
    tuple val(meta), path("*_pathways.tsv"), emit: kegg_pathways
    path "versions.yml"                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def kegg_input = ko_list ? "-l ${ko_list}" : "-i ${filtered_tbl}"

    if (ko_list && filtered_tbl){
        log.warn("Both \$ko_list and \$filtered_tbl were given as input types, will fall back to using \$ko_list i.e. ${ko_list}");
    }

    """
    give_pathways \\
    ${kegg_input} \\
    -o ${prefix}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        kegg-pathways-completeness: \$(give_pathways --version | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}.kegg_contigs.tsv
    touch ${prefix}.kegg_pathways.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        kegg-pathways-completeness: \$(give_pathways --version | cut -d' ' -f2)
    END_VERSIONS
    """
}
