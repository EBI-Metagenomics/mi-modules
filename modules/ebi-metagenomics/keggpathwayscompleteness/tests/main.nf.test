
nextflow_process {

    name "Test Process KEGGPATHWAYSCOMPLETENESS"
    script "../main.nf"
    process "KEGGPATHWAYSCOMPLETENESS"

    tag "modules"
    tag "modules_nfcore"
    tag "keggpathwayscompleteness"

    test("keggpathwayscompleteness - hmmsearch_table") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test.fasta", checkIfExists: true)
                    ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test.tbl", checkIfExists: true)
                    ]
                input[2] = [
                    [ id:'test', single_end:false ], // meta map
                    [] // empty list for optional input
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("keggpathwayscompleteness - KO list") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [] // empty list for optional input
                    ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    [] // empty list for optional input
                    ]
                input[2] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test_kos.txt", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("keggpathwayscompleteness - stub") {

        options "-stub"

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test.fasta", checkIfExists: true)
                    ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test.tbl", checkIfExists: true)
                    ]
                input[2] = [
                    [ id:'test', single_end:false ], // meta map
                    [] // empty list for optional input
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
