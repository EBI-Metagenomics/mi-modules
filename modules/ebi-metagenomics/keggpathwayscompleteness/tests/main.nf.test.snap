{
    "keggpathwayscompleteness - filtered_tbl": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.summary.kegg_contigs.tsv:md5,a31fd7add583c5f29f39144d2009948f"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.summary.kegg_pathways.tsv:md5,44d303bdd75a8965ab6b519f8ab32db2"
                    ]
                ],
                "2": [
                    "versions.yml:md5,fdc86d119def3d1ab9b14546eb3efe4a"
                ],
                "kegg_contigs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.summary.kegg_contigs.tsv:md5,a31fd7add583c5f29f39144d2009948f"
                    ]
                ],
                "kegg_pathways": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.summary.kegg_pathways.tsv:md5,44d303bdd75a8965ab6b519f8ab32db2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fdc86d119def3d1ab9b14546eb3efe4a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-07-09T09:44:04.378610297"
    },
    "keggpathwayscompleteness - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.kegg_contigs.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.kegg_pathways.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,fdc86d119def3d1ab9b14546eb3efe4a"
                ],
                "kegg_contigs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.kegg_contigs.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "kegg_pathways": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.kegg_pathways.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fdc86d119def3d1ab9b14546eb3efe4a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-07-09T09:25:11.02773529"
    },
    "keggpathwayscompleteness - KO list": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.summary.kegg_contigs.tsv:md5,d18735f58cc8593bd7abc0270e1af891"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.summary.kegg_pathways.tsv:md5,e6538c3dd00b1ffb015780a90003f81d"
                    ]
                ],
                "2": [
                    "versions.yml:md5,fdc86d119def3d1ab9b14546eb3efe4a"
                ],
                "kegg_contigs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.summary.kegg_contigs.tsv:md5,d18735f58cc8593bd7abc0270e1af891"
                    ]
                ],
                "kegg_pathways": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.summary.kegg_pathways.tsv:md5,e6538c3dd00b1ffb015780a90003f81d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fdc86d119def3d1ab9b14546eb3efe4a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-07-09T09:25:03.771670203"
    }
}